darts/ad/utils.py

```diff
@@ -421,106 +434,77 @@ def show_anomalies_from_scores(
+        fig, axs = plt.subplots(
+            plots_per_ts,
+            figsize=(8, 4 + 2 * (plots_per_ts - 1)),
+            sharex=True,
+            gridspec_kw={"height_ratios": [2] + [1] * (plots_per_ts - 1)},
+            squeeze=False,
         )
 
-    axs[index_ax][0].set_title("")
-
-    if anomalies is not None or pred_scores is not None:
-        axs[index_ax][0].set_xlabel("")
-
-    axs[index_ax][0].legend(loc="upper center", bbox_to_anchor=(0.5, 1.1), ncol=2)
-
-    if pred_scores is not None:
-        dict_input = {}
-
-        for idx, (score, w) in enumerate(zip(pred_scores, window)):
-            dict_input[idx] = {"series_score": score, "window": w, "name_id": idx}
-
-        for index, elem in enumerate(
-            sorted(dict_input.items(), key=lambda x: x[1]["window"])
-        ):
-            if index == 0:
-                current_window = elem[1]["window"]
-                index_ax = index_ax + 1
-
-            idx = elem[1]["name_id"]
-            w = elem[1]["window"]
-
-            if w != current_window:
-                current_window = w
-                index_ax = index_ax + 1
-
-            if metric is not None:
-                value = round(
-                    eval_metric_from_scores(
-                        anomalies=anomalies,
-                        pred_scores=pred_scores[idx],
-                        window=w,
-                        metric=metric,
-                    ),
-                    3,
-                )
-            else:
-                value = None
-
-            if names_of_scorers is not None:
-                label = names_of_scorers[idx] + [f" ({value})", ""][value is None]
-            else:
-                label = f"score_{str(idx)}" + [f" ({value})", ""][value is None]
-
-            _plot_series(
-                series=elem[1]["series_score"],
-                ax_id=axs[index_ax][0],
-                linewidth=0.5,
-                label_name=label,
-            )
-
-            axs[index_ax][0].legend(
-                loc="upper center", bbox_to_anchor=(0.5, 1.19), ncol=2
+    if multivariate_plot:
+        if pred_series is not None:
+            pred_series = _check_input(
+                pred_series,
+                name="pred_series",
+                width_expected=series.width,
+                check_multivariate=True,
+            )[0]
+
+        if anomalies is not None:
+            anomalies = _check_input(
+                anomalies,
+                name="anomalies",
+                width_expected=series.width,
+                check_binary=True,
+                check_multivariate=True,
+            )[0]
+
+        if pred_scores is not None:
+            for pred_score in pred_scores:
+                pred_score = _check_input(
+                    pred_score,
+                    name="pred_score",
+                    width_expected=series.width,
+                    check_multivariate=True,
+                )[0]
```

Comment:
these checks could actually also be conducted when `multivariate_plot=False`, and also check `num_series_expected` while we are at it.